.github/workflows/build.yml

```diff
@@ -39,18 +40,30 @@ jobs:
         with:
           submodules: true
 
-      - name: compile
-        uses: KSPModdingLibs/KSPBuildTools/.github/actions/compile@main
+      - uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 5.x
+
+      # Install CKAN and set up an instance. If CKAN is not needed, remove this step
+      - uses: KSPModdingLibs/KSPBuildTools/.github/actions/ckan-setup@main
         with:
-          build-configuration: ${{ inputs.build-configuration }}
-          ksp-zip-url: ${{ inputs.ksp-zip-url }}
-          ksp-zip-password: ${{ secrets.ksp-zip-password }}
           dependency-identifiers: ${{ inputs.dependency-identifiers }}
+
+      # Install any CKAN dependencies listed in the msbuild file. If there are none, remove this step
+      - uses: KSPModdingLibs/KSPBuildTools/.github/actions/install-dependencies@main
```

Comment:
The reason that all of the dependency inputs were passed to the compile action is that it never makes sense to compile without installing dependencies first.  Modularity is nice, but breaking them out now makes these workflows more verbose.  We could of course offer an action that just does both.  Modularity may also have an advantage if you can somehow cache the build environment (dependencies, ksp libs) so that you only need to run the compile action.